### ISSUE TITLE
Switching between controls leaves single point shapes behind

### ISSUE THREAD

**Issue by iandanforth** [Leaflet/Leaflet.Editable#62]
In the basic controls example if you select the rectangle control and then click to another control, you'll leave a single point rectangle behind the second control.

Ideally there would be a check to make sure drawing cannot start when the event is fired over any leaflet control.

onMouseDown in L.Editable may be a good place for this. 

Maybe something like http://stackoverflow.com/a/3028037/1775741 that looks for 'leaflet-control'

**Comment by yohanboniface** [comment]
Yes, those controls are very very basic, just for the demo, I'll take care on stopping click propagation (in a mission in Lesbos right now, will be back on affairs in roughly one week).

**Comment by iandanforth** [comment]
I've tried to hack around this, but I'm having a hard time cleaning up the layers being drawn. Ideally I want to do this

1. Click a control to start drawing
2. Click again to place the marker/rectangle/circle start point on the map
  2.5 If this second click is within a leaflet-control, remove any layers created in 1.

I've added in checks for 2.5 in L.Editable.onMouseUp but I don't know how to access the layer(s) that were already created in 1 and properly clean them up! Calling stopDrawing(); doesn't work for example.
